File: rust/src/models/ingest_failure.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct IngestFailure {
    pub event_id: String,

    pub reason: String,
}

impl IngestFailure {
    pub fn new(event_id: String, reason: String) -> Self {
        Self { event_id, reason }
    }
}

File: typescript/src/models/ingestFailure.ts
// this file is @generated

export interface IngestFailure {
  eventId: string;

  reason: string;
}

export const IngestFailureSerializer = {
  _fromJsonObject(object: any): IngestFailure {
    return {
      eventId: object["event_id"],
      reason: object["reason"],
    };
  },

  _toJsonObject(self: IngestFailure): any {
    return {
      event_id: self.eventId,
      reason: self.reason,
    };
  },
};
